### ISSUE TITLE
Multiple fragments with same index fire only once

### ISSUE THREAD

**Issue by mlange-42** [hakimel/reveal.js#3087]
With multiple fragments of the same `data-fragment-index` on a slide, only the first one fires events `fragmentshown` and `fragmenthidden`. This is counter-intuitive given the event names.

## Steps to reproduce

Fragments on slide:

```html
<p class='fragment' data-fragment-index='1'>Fragment 1</p>
<p class='fragment' data-fragment-index='1'>Fragment 2</p>
```

JS:

```js
Reveal.on( 'fragmentshown', event => {
    console.log('showing '+event.fragment.textContent);
} );
```

Prints only:
```
Fragment 1
```

Tested on Windows 10; Firefox 92.0 and Chrome 94.0.

**Comment by josharsh** [comment]
@mlange-42  I might be wrong here, but do you think this is because of the same value for "data-fragment-index"?

**Comment by mlange-42** [comment]
@josharsh Yes, this is exactly what I mean. Of multiple fragments showing in the same "step" (i.e. same `data-fragment-index`), I receive an event only for the first one. In my code, I thus have to check the slide for other fragments with the same index. See [reveal-svg-smil.js#L139](https://github.com/mlange-42/reveal-svg-smil/blob/6f214fc1a3330d7d5fc674dc77abb3b366a30aa5/reveal-svg-smil.js#L139).

This would be expected if the signal name was `fragmentindexshown`, but with `fragmentshown` I would expect an event per fragment entry.
